app/controllers/registered_applications_controller.rb

```diff
@@ -4,6 +4,11 @@ def index
   end
 
   def show
+    begin
+      @application = current_user.applications.find(params[:id])
+    rescue ActiveRecord::RecordNotFound
```

Comment:
Its usually OK to let the `ActiveRecord::RecordNotFound` error to bubble upward as Rails will translate this to a 404 page to the user which is fine as they are typically requesting a URL that does not exist as far as they are concerned.